### ISSUE TITLE
Can't install 3.9.1 on M1 with GCC 12: "error: implicit declaration of function '__builtin_flt_rounds'"

### ISSUE THREAD

**Issue by Brodan** [pyenv/pyenv#2509]
I've seen a lot of stuff saying 3.9.0+ has no issues on M1 but I have absolutely not found this to be the case. I'm trying to install 3.9.1 via pyenv with rosetta 2/brew. This is the command I'm running:

`CC="$(brew --prefix gcc)/bin/gcc-12" pyenv install 3.9.1 -v`

### Description
- [x] Platform information (e.g. Ubuntu Linux 16.04): M1 Mac (Monterey 12.6.1)
- [x] OS architecture (e.g. amd64): arm64 but using Rosetta 2
- [x]  pyenv version: `2.3.5
- [x] Python version: `3.9.1`
- [x] C Compiler information (e.g. gcc 7.3): `gcc: stable 12.2.0`
- [x] Xcode 14.1, 
- [x] homebrew version: 3.6.8
- [x] Please attach the debug trace of the failing command as a gist:

This result in an error waaaay down in the process that looks as such:

```
In file included from ./Include/tupleobject.h:39,
                 from ./Include/Python.h:105:
Objects/floatobject.c: In function 'PyFloat_GetInfo':
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/float.h:76:21: error: implicit declaration of function '__builtin_flt_rounds'; did you mean '__builtin_llround'? [-Werror=implicit-function-declaration]
   76 | #define FLT_ROUNDS (__builtin_flt_rounds())
      |                     ^~~~~~~~~~~~~~~~~~~~
./Include/cpython/tupleobject.h:30:69: note: in definition of macro 'PyTuple_SET_ITEM'
   30 | #define PyTuple_SET_ITEM(op, i, v) (_PyTuple_CAST(op)->ob_item[i] = v)
      |                                                                     ^
Objects/floatobject.c:90:5: note: in expansion of macro 'PyStructSequence_SET_ITEM'
   90 |     PyStructSequence_SET_ITEM(floatinfo, pos++, PyLong_FromLong(flag))
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~
Objects/floatobject.c:104:5: note: in expansion of macro 'SetIntFlag'
  104 |     SetIntFlag(FLT_ROUNDS);
      |     ^~~~~~~~~~
Objects/floatobject.c:104:16: note: in expansion of macro 'FLT_ROUNDS'
  104 |     SetIntFlag(FLT_ROUNDS);
      |                ^~~~~~~~~~
```

Has anyone else experienced this or figured out a workaround? I have no idea where to turn.

**Comment by xrisk** [comment]
Is there a good reason for settings the compiler to gcc explicitly? `__builtin_flt_rounds()` is a clang instrinsic.

**Comment by native-api** [comment]
The immediate cause is that we include XCode SDK into `INCLUDE_PATH` -- which breaks compilers incompatible with it.

But then the compilation still fails due to a flaw in CPython's `Modules/posixmodule.c`. That flaw is unfixed even in `main`, CPython was clearly not tested in MacOS with GCC.

While we can fix and patch all that... should we really support building CPython with GCC in MacOS of the upstream clearly doesn't?

**Comment by Brodan** [comment]
> Is there a good reason for setting the compiler to gcc explicitly? `__builtin_flt_rounds()` is a clang instrinsic.

If i don't set it explicity, then I get an error much earlier on in the install process:
```
checking for c++... c++
configure:

  By default, distutils will build C++ extension modules with "c++".
  If this is not intended, then set CXX on the configure command line.
  
checking for the platform triplet based on compiler characteristics... darwin
configure: error: internal configure error for the platform triplet, please file a bug report
make: *** No targets specified and no makefile found.  Stop.

BUILD FAILED (OS X 12.6.1 using python-build 2.2.2-6-g28628580)
```

**Comment by native-api** [comment]
> checking for the platform triplet based on compiler characteristics... darwin

That's strange. 3.9.1 does support XCode 13.3+, this shouldn't be happening.

Do you by any chance have Homebrew's LLVM on `PATH`?

If not, please provide the debug trace of `pyenv install` as per the usse template and also `config.log` from the build directory.

**Comment by native-api** [comment]
> checking for the platform triplet based on compiler characteristics... darwin

That's strange. We patch 3.9.1 to support XCode 13.3+, this shouldn't be happening.

Please check:

* If you have another Pyenv on PATH that the one you intend to use.
* If you have Homebrew's LLVM on `PATH`

If not, please provide the debug trace of `pyenv install` as per the usse template and also `config.log` from the build directory.

**Comment by Brodan** [comment]
> > checking for the platform triplet based on compiler characteristics... darwin
> 
> That's strange. We patch 3.9.1 to support XCode 13.3+, this shouldn't be happening.
> 
> Please check:
> 
>     * If you have another Pyenv on `PATH` than the one you intend to use.
> 
>     * If you have Homebrew's LLVM on `PATH`
> 
> 
> If not, please provide the debug trace of `pyenv install` as per the usse template and also `config.log` from the build directory.

Thanks for troubleshooting with me. I double checked and cleaned up my `PATH` and same issue.

My `PATH` currently has the following in it:
```
'/opt/homebrew/bin',
'/opt/homebrew/sbin',
'/usr/local/bin',
'/usr/bin',
'/bin',
'/usr/sbin',
'/sbin',
'/usr/local/go/bin',
'/Library/Apple/usr/bin',
'/opt/homebrew/'
```

I checked these directories and don't see an LLVM in any of the Homebrew dirs. I checked for LLVM in my path specifically and only get this:

```
$ which llvm-gcc 
/usr/bin/llvm-gcc
```

Same with brew and pyenv:

```
$ which pyenv
/opt/homebrew/bin/pyenv
$ which brew
/opt/homebrew/bin/brew
$ brew list
[python-build.20221103174811.34271.log](https://github.com/pyenv/pyenv/files/9933220/python-build.20221103174811.34271.log)

==> Formulae
autoconf	ca-certificates	gcc		gmp		isl		libmpc		lz4		m4		mpfr		openssl@1.1	pkg-config	pyenv		readline	xz		zstd
```

Attached is the log file I got after running pyenv with the verbose and debug flags (`pyenv install 3.9.1 -v -g `).

[python-build.20221103174811.34271.log](https://github.com/pyenv/pyenv/files/9933222/python-build.20221103174811.34271.log)

**Comment by Brodan** [comment]
So I ended up nuking all my homebrew stuff and starting from scratch and eventually got everything sorted it seems. I had an x86 brew alias with a non-convential `HOMEBREW_PREFIX` setting that left a bunch of packages in my path that I didn't realize existed. I uninstalled everything, installed Homebrew fresh without Rosetta 2, and I was able to install `3.9.1` with only one warning. Thanks for helping me troubleshoot.
